### ISSUE TITLE
`Cannot find module 'next/dist/next-server/server/node-polyfill-fetch'` for vercel monorepo serverless functions

### ISSUE THREAD

**Issue by shunkakinoki** [vercel/next.js#16667]
# Bug report

Hi, first of all love your work!

I'm migrating to use the NEW monorepo support for vercel,

and although the functions succeed to deploy, I get the following error when invoking it.
## Describe the bug

`Cannot find module 'next/dist/next-server/server/node-polyfill-fetch'`

## To Reproduce

Reproduction coming ...

## System information

- OS: [e.g. macOS, Windows]
- Browser (if applies) [e.g. chrome, safari]
- Version of Next.js: [e.g. 6.0.2]
- Version of Node.js: [e.g. 10.10.0]

**Comment by timneutkens** [comment]
Please provide a full reproduction.

**Comment by belgattitude** [comment]
@timneutkens I've contacted support already, but you test it here:

- https://github.com/belgattitude/vercel-monorepo-test
- https://vercel-monorepo-test-web-app.vercel.app/

**Comment by vongohren** [comment]
Just adding this, 71day old issue aswell: https://github.com/vercel/next.js/discussions/14405

**Comment by vongohren** [comment]
Im also experiencing this on all my deployes, both production and test environemnt

**Comment by mattphillips** [comment]
Sorry for the +1 spam on this but unfortunately I am also running into this issue.

I have a workaround for this issue (that I had been using previously when monorepo support was experimental) if anyone finds it useful:

I am setting the `Root Directory` to the yarn workspace and have a custom build command that builds the Next.js app and copies all of the compiled code up into the workspace for deployment. Example here: https://github.com/mattphillips/next-public

This solution isn't great but I'm unable to upgrade to full monorepo support on Vercel while all functions are falling over with this bug.

**Comment by vongohren** [comment]
@mattphillips Im not able to use this workaround to deploy new versions. @Timer is this beeing investigated because my production is currently broken and I cannot deploy a new version, because this happens each time.

**Comment by paulogdm** [comment]
The team is taking a look as we speak. It should be resolved in a few days.

Our apologies for the inconvenience.

**Comment by vongohren** [comment]
@paulogdm any news? 😊

**Comment by good-idea** [comment]
If it's any help, here are two repos that have the same setup, one is working while the other is not:

https://github.com/good-idea/hikawa.studio (does not work - the app builds & prerenders SSG successfully, but requests to all pages fail with the same error as above)
https://github.com/good-idea/spinellikilcollin.com (works)

Both have the following vercel settings:

![image](https://user-images.githubusercontent.com/11514928/92181247-ad67ef00-edfd-11ea-8dee-17ea678e668c.png)

**Comment by JakeCooper** [comment]
Not sure it'll help, but here's another project with the same issue. Blocking production deployments sadly as well :(

https://vercel.com/railwayapp/railway-web

**Comment by timneutkens** [comment]
We've tracked down the issue yesterday and the team is working on a fix 🙏

**Comment by belgattitude** [comment]
@timneutkens that' fantastic ! You rock guys. Thanks a million.

**Comment by JakeCooper** [comment]
+1 to what @belgattitude said! Great work, and loving the official monorepo support so far!

**Comment by paulogdm** [comment]
Hello everyone 👋 
A fix has been shipped to canary (`@vercel/next@canary`). I will post updates on how to try it out here.

**Comment by paulogdm** [comment]
You can test the fix by following these steps:

- Include the following environment variable in your deployment (through the [environment variables UI](https://vercel.com/docs/build-step#environment-variables)): Key: `FORCE_BUILDER_TAG`. Value: `canary`
![image](https://user-images.githubusercontent.com/7690649/92308134-e32de480-ef71-11ea-96d3-eb1f81fae06a.png)

- If you have a `vercel.json` file, then you can follow the same method:
```json
{
  "build": {
    "env": {
      "FORCE_BUILDER_TAG": "canary"
    }
  }
}
```

Let us know if you have positive results.

**Comment by belgattitude** [comment]
Hey @paulogdm I can confirm it fix the issue with api-route: https://github.com/belgattitude/vercel-monorepo-test/pull/10 with api routes. 

Great work !

PS: any idea why a simple api is so slow ?: 
- link: https://vercel-monorepo-test-web-app-git-vercelforcebuildertag.belgattitude.vercel.app/api/hello (200/300ms)
- source: https://github.com/belgattitude/vercel-monorepo-test/blob/master/apps/web-app/src/pages/api/hello.ts

**Comment by paulogdm** [comment]
> PS: any idea why a simple api is so slow ?:

I would recommend that you check the region where you deployed your function. It seems to be in `SFO1`. Therefore, the response you are observing is due to latency (probably).

**Comment by vongohren** [comment]
@paulogdm Im still getting the error for some reason. Trying out both the env variable in the settings and tried the vercel.json. Nothing positive happens to me unfortunately. Same logs

```
00:23:20:36
2020-09-06T22:23:20.479Z	8438ba21-b91f-4ac2-8bab-adcd27423b7b	ERROR	Unhandled error during request: Error: Cannot find module 'next/dist/next-server/server/node-polyfill-fetch'
Require stack:
- /var/task/build/serverless/pages/api/apartment/[id].js
- /var/task/now__launcher.js
- /var/runtime/UserFunction.js
- /var/runtime/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
    at Function.Module._load (internal/modules/cjs/loader.js:842:27)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at eval (webpack-internal:///mctB:1:18)
    at Object.mctB (/var/task/build/serverless/pages/api/apartment/[id].js:239:1)
    at __webpack_require__ (/var/task/build/serverless/pages/api/apartment/[id].js:23:31)
    at eval (webpack-internal:///iWDc:4:106)
    at Module.iWDc (/var/task/build/serverless/pages/api/apartment/[id].js:210:1)
    at __webpack_require__ (/var/task/build/serverless/pages/api/apartment/[id].js:23:31) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/var/task/build/serverless/pages/api/apartment/[id].js',
    '/var/task/now__launcher.js',
    '/var/runtime/UserFunction.js',
    '/var/runtime/index.js'
  ]
}
2020-09-06T22:23:20.479Z	8438ba21-b91f-4ac2-8bab-adcd27423b7b	ERROR	Uncaught Exception 	{"errorType":"Error","errorMessage":"Cannot find module 'next/dist/next-server/server/node-polyfill-fetch'\nRequire stack:\n- /var/task/build/serverless/pages/api/apartment/[id].js\n- /var/task/now__launcher.js\n- /var/runtime/UserFunction.js\n- /var/runtime/index.js","code":"MODULE_NOT_FOUND","requireStack":["/var/task/build/serverless/pages/api/apartment/[id].js","/var/task/now__launcher.js","/var/runtime/UserFunction.js","/var/runtime/index.js"],"stack":["Error: Cannot find module 'next/dist/next-server/server/node-polyfill-fetch'","Require stack:","- /var/task/build/serverless/pages/api/apartment/[id].js","- /var/task/now__launcher.js","- /var/runtime/UserFunction.js","- /var/runtime/index.js","    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)","    at Function.Module._load (internal/modules/cjs/loader.js:842:27)","    at Module.require (internal/modules/cjs/loader.js:1026:19)","    at require (internal/modules/cjs/helpers.js:72:18)","    at eval (webpack-internal:///mctB:1:18)","    at Object.mctB (/var/task/build/serverless/pages/api/apartment/[id].js:239:1)","    at __webpack_require__ (/var/task/build/serverless/pages/api/apartment/[id].js:23:31)","    at eval (webpack-internal:///iWDc:4:106)","    at Module.iWDc (/var/task/build/serverless/pages/api/apartment/[id].js:210:1)","    at __webpack_require__ (/var/task/build/serverless/pages/api/apartment/[id].js:23:31)"]}
Unknown application error occurred
```

**Comment by paulogdm** [comment]
@vongohren Do you have a minimal repro that we can investigate?

**Comment by vongohren** [comment]
@paulogdm yes yes, sorry for late reply! But here are two links to follow

https://github.com/vongohren/vercel-function-error
https://repro-error.vercel.app/

When clicking google sign in it sends a get function and gets 502 back. The error in the logs are this
<img width="1226" alt="Screenshot 2020-09-09 at 01 23 00" src="https://user-images.githubusercontent.com/1012055/92536872-0b9c3400-f23b-11ea-9ede-334962296f6d.png">

**Comment by styfle** [comment]
@vongohren It looks like you used the Override switch to change your project settings Output Directory. Try using the default settings instead of Override.

**Comment by Timer** [comment]
Closing this as we confirmed a few of the reproductions above are now fixed.

**Comment by vongohren** [comment]
@styfle thanks for reply, so what you are telling me that there is a bug in vercel when using this setup and override? Or is there anythin else I can do differently?

@Timer did you guys confirm test my case? And that was the override situation?

**Comment by styfle** [comment]
@vongohren I confirmed [your code](https://github.com/vongohren/vercel-function-error) deploys fine without using any Override switches, just using defaults.

Your code uses `next build` which sets the default output directory to `.next` but your projects settings was attempting to use a different output directory, `build`.

https://github.com/vongohren/vercel-function-error/blob/91635ad4f549d5a1dc85af124c8dd1d51b1e768e/package.json#L44

The solution is to uncheck the Override option in your project settings, thanks!

**Comment by vongohren** [comment]
@styfle thanks for quick replies, but im sorry, this does not work for the current base im trying on.
I turned off all three overrides, made sure that my build, the next config does not choose a dist dir, in my code.
Deployed a new production build, and still experiencing it. Do I need to create a new project or what?

<img width="1324" alt="Screenshot 2020-09-12 at 23 53 43" src="https://user-images.githubusercontent.com/1012055/93005666-73ac8c00-f553-11ea-8b6d-6ece716ad62c.png">

**Comment by vongohren** [comment]
I can confirm removing the https://github.com/vongohren/vercel-function-error/blob/master/next.config.js.backup without the backup, makes it work again.

Things was git ignored so it was lost in context, sorry about that.
But then atleast we have the culprit and maybe you guys can have some insight to letting me know why it failed?

**Comment by yournatalita** [comment]
@vongohren Thank you for the comment! You gave me a hint that I maybe have the same issue. Comparing with your file I found out that line `config.devtool = 'eval-source-map';` is causing this error on Vercel in my case. 

Removing it or adding conditional dev/build check helped for me.

Even though I resolved my issue, I haven't found any docs about it and in my custom server project I haven't had the issue with source-map enabled.

I find it handy on dev servers to be able to inspect scripts. Is this bug? Or designed behaviour? 
Where can I read the info about webpack settings that should be blocked on serverless vercel deploy?
